Core/src-test/com/serotonin/m2m2/web/mvc/spring/security/authentication/MangoTokenAuthenticationProviderTest.java

```diff
@@ -0,0 +1,101 @@
+package com.serotonin.m2m2.web.mvc.spring.security.authentication;
+
+import static com.serotonin.m2m2.Common.getBean;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.junit.Assert.assertThrows;
+
+import java.time.Duration;
+import java.util.Date;
+import java.util.stream.Collectors;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.CredentialsExpiredException;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken;
+import org.springframework.security.web.authentication.WebAuthenticationDetails;
+
+import com.infiniteautomation.mango.spring.components.TokenAuthenticationService;
+import com.infiniteautomation.mango.spring.service.UsersService;
+import com.serotonin.m2m2.MangoTestBase;
+import com.serotonin.m2m2.MockMangoLifecycle;
+import com.serotonin.m2m2.vo.User;
+import com.serotonin.timer.SimulationTimer;
+
+import io.jsonwebtoken.ExpiredJwtException;
+
+/**
+ * @author Jared Wiltshire
+ */
+
+public class MangoTokenAuthenticationProviderTest extends MangoTestBase {
+
+  AuthenticationManager authenticationManager;
+  TokenAuthenticationService tokenAuthService;
+  UsersService usersService;
+  SimulationTimer clock;
+
+  @Override
+  protected MockMangoLifecycle getLifecycle() {
+    return super.getLifecycle().setInitializeWebServer(true);
```

Comment:
Does this method (`setInitializeWebServer`) actually exist in 4.5.x?

Reply:
@jazdw yes it does, I was able to run successfully the test before these changes.